### PR TITLE
Feat/62 password visibility toggle

### LINKED ISSUE #62: [FEATURE] 🗺️ ⚙️ Settings Pages > Password Settings > implement a password visibility toggle
figma link - https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=10-17260&t=ybfmYNMB6xh8HxXT-1

### **Description**
Develop a Password Visibility Toggle component for the Settings Pages > Password Settings section. This component will allow users to toggle the visibility of their password input. The component should be user-friendly and easy to integrate into existing forms. It should include the following features:

### **Acceptance Criteria**

The Password Visibility Toggle component is implemented according to the design.
The component should allow users to toggle the password input type between "password" and "text".
Ensure the component is fully responsive and works well across different screen sizes.
Ensure accessibility standards are met, including appropriate ARIA roles and labels.
Requirements
The component should receive the following props:

### **password: The user-entered password.**
onPasswordChange: A callback function that is triggered when the password changes.
### **Testing**

Ensure the component correctly toggles the password visibility.
Check that the component handles user interactions appropriately.
Test the component for responsiveness and accessibility.
Verify that the onPasswordChange callback is triggered with the correct password value.

**### Tasks**

- [ ] Develop the Password Visibility Toggle component based on the provided requirements.
- [ ] Implement handling for toggling the password input type.
- [ ] Ensure proper keyboard navigation and ARIA roles for accessibility.
- [ ] Test the component for responsiveness and compatibility across different devices.

Frame 1321315355

### PR DESCRIPTION
**Pull request by Praizee**
# Description

**Closes #62**

# Changes proposed

## What were you told to do?

[FEATURE] 🗺️ ⚙️ Settings Pages > Password Settings > implement a password visibility toggle

**Figma link**: https://www.figma.com/design/VEItfX6St5NSAqqNHImcxD/HNG-Boilerplate-Designs?node-id=10-17260&t=ybfmYNMB6xh8HxXT-1

## What did you do?
- Installed shadcn input component
- Created an 'svg' folder within an 'assets' folder in the 'public' folder for the SVG icons: EyeOpen and EyeClose
- Implemented a responsive password visibility toggle component in the 'common' components folder, meeting accessibility standards based on the provided requirements
- Included a demo component to show the password component's usage
- Wrote unit tests to:
  - Ensure the component correctly toggles the password visibility.
  - Check that the component handles user interactions appropriately.
  - Test the component for responsiveness and accessibility.
  - Verify that the onPasswordChange callback is triggered with the correct password value.

# Check List (Check all the applicable boxes)

🚨Please review the [contribution guideline](CONTRIBUTING.md) for this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/Videos

image

image

image

image

### CODE REVIEW

File: src/components/ui/input.tsx [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import * as React from "react"
```
Comment (Justfemi) [inline src/components/ui/input.tsx]: Use the Input component for consistency around the website

**Review by Justfemi — CHANGES_REQUESTED** [review]

File: src/components/ui/input.tsx (hunk 1 above)
Comment (Praizee) [reply src/components/ui/input.tsx]: Which?
I used the input from shadcn

Comment (Praizee) [reply src/components/ui/input.tsx]: @Justfemi 
Someone's created the input component in a PR. I've pulled from upstream dev and merged already

**Comment by Praizee** [comment]
@JulianaSau 
The requested changes have been addressed

**Comment by taiwonaf** [comment]
- Fix the conflict and make another pr

**Pull request closed without merging**